tap_bing_ads/__init__.py

```diff
@@ -530,6 +529,17 @@ def get_report_metadata(report_name, report_schema):
         lambda field: metadata_fn(report_name, field, required_fields),
         report_schema['properties']))
 
+def snakecase(string):
+    string = re.sub(r"[\-\.\s]", '_', str(string))
+    if not string:
+        return string
+    return (string[0]).lower() + re.sub(r"[A-Z]", lambda matched: '_' + (matched.group(0)).lower(), string[1:])
+
+def pascalcase(string):
+    string = snakecase(string)
```

Comment:
Two passes over `string` seems a little wasteful.

If you add a new function to do the word splitting, then you can reuse it in `snakecase` and `pascalcase`.

```python

def make_words(string):
    pass

def snakecase(string):
    return "_".join(word.lower() for word in make_words(string))

def pascalcase(string):
    return "".join(word.title() for word in make_words(string))
```

Reply:
I was trying to keep all the functionality of the old library, which gets a little tricky with one pass over splitting. However, the use case in the tap is only for CamelCase hardcoded stream names to snake_case and back. I'll update the function to something simpler that fits that case.